doctr/utils/data.py

```diff
@@ -84,8 +89,14 @@ def download_from_url(
         logging.info(f"Using downloaded & verified file: {file_path}")
         return file_path
 
-    # Create folder hierarchy
-    folder_path.mkdir(parents=True, exist_ok=True)
+    try:
+        # Create folder hierarchy
+        folder_path.mkdir(parents=True, exist_ok=True)
+    except OSError:
+        print(f"""
+        Failed creating cache direcotry: {cache_dir}
```

Comment:
minor typo -> directory @frgfm @odulcy-mindee are you ok with printing instead of logging in this case ?

Reply:
I would suggest to raise the exception here because if the function is unable to create the cache dir, it probably means that the python script lacks permissions to create the folder or the disk is full or any other I/O Error (`exist_ok=True` so it will not raise if the folder already exists) so there is no need to try to download and save files because it'll fail.

What about:

```python
    except OSError as e:
        error_message = f"Failed creating cache direcotry at {folder_path}"
        if DOCTR_CACHE_DIR_ENV_NAME in os.environ:
            error_message += f" using path from {DOCTR_CACHE_DIR_ENV_NAME} environment variable."
        else:
            error_message += f". You can change default cache directory using {DOCTR_CACHE_DIR_ENV_NAME} environment variable if needed."
        raise OSError(error_message) from e
```

Also note it should be `folder_path` and not `cache_dir` in error message

Reply:
About information display, I would either go for exception raising or logging, but no print. Logging level can be set by the user as they wish, exception can be caught, `print` doesn't have that flexibility :sweat_smile: